aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1249,6 +1316,51 @@ private boolean s3KeyIdAccessKeyBothConfigured() {
     return (s3AccessKeyId == null) == (s3SecretAccessKey == null);
   }
 
+  private AwsCredentialsProvider credentialsProvider(String credentialsProviderClazz) {
+    try {
+      Class<?> providerClass =
+          DynClasses.builder()
+              .loader(this.getClass().getClassLoader())
+              .impl(credentialsProviderClazz)
+              .buildChecked();
+
+      ValidationException.check(
+          AwsCredentialsProvider.class.isAssignableFrom(providerClass),
+          "%s is not an instance of software.amazon.awssdk.auth.credentials.AwsCredentialsProvider, loaded by the classloader for %s",
```

Comment:
Error message format in iceberg is 
"Cannot X because Y. (Remediation)"

Also in this I would probably embed the class name as a variable in the format string. "AwsCredentialsProvider.class.getName"

Reply:
Fixed all exception messages as recommended. thanks